src/datasources.js

```diff
@@ -1,3 +1,45 @@
+/**
+ * Asynchronously loads data sources from vellum.opts().core.dataSources
+ * Currently only supports fixtures
+ *
+ * Format in opts:
+ * dataSources: [
+ *     {
+ *          key: string : data source name (fixture, case, products...)
+ *          endpoint: function or string (URL)
+ *     }
+ * ]
+ *
+ * The endpoint function (or URL) should return the following format:
+ * [
+ *      {
+ *          sourceUri: string (used in the instance definition)
+ *          defaultId: string (used in instance definition)
+ *          intialQuery: string (used in nodeset)
+ *          name: string (text in the dropdown)
+ *          structure: nested dictionary of attributes
```

Comment:
Since this is describing XML, and in XML "attributes" are different from "elements", it might make more sense to say "nested dictionary describing the data source"